log-analyzer/features/org.wso2.carbon.la.apim.feature/src/main/resources/gadgets/LogAPIMArtifactDeleted/js/VizGrammar.js

```diff
@@ -0,0 +1,2300 @@
+                }
+            }
+        };
+    }
+
+    return mark;
+}
+
+;var map = function(dataTable, config) {
+
+    this.metadata = dataTable[0].metadata;
+    var marks ;
+    var signals ;
+    var predicates = [];
+    var legends = [];
+    this.spec = {};
+    var geoInfoJson ;
+
+    geoInfoJson = loadGeoMapCodes(config.helperUrl);
+    config = checkConfig(config, this.metadata);
+    this.config = config;
+    this.config.geoInfoJson = geoInfoJson;
+
+    for (i = 0; i < dataTable[0].values.length; i++) {
+        for (var key in dataTable[0].values[i]) {
+            if(key == dataTable[0].metadata.names[config.x]){
+                if (dataTable[0].values[i].hasOwnProperty(key)) {
+                    dataTable[0].values[i].unitName = dataTable[0].values[i][key];
+                    dataTable[0].values[i][key] = getMapCode(dataTable[0].values[i][key], config.mapType,geoInfoJson);
+                    break;
+                }
+            }
+        }
+    };
+
+    dataTable[0].name = config.title;
+    dataTable[0].transform = [
+        {
+            "type": "formula",
+            "field": "v",
+            "expr": "datum."+this.metadata.names[config.y]
+        }
+    ];
+
+    if (config.tooltip.enabled) {
+        marks = getMapMark(config, this.metadata);
+        signals = getMapSignals();
+        this.spec.signals = signals;
+    }
+
+    dataTable.push(getTopoJson(config,this.metadata));
+    predicates.push(getMapPredicates());
+    legends.push(getMapLegends(config,this.metadata));
+
+    var cScale = {
+        "name": "color",
+        "type": "linear",
+        "domain": {"data": "geoData","field": "zipped.v"},
+        "domainMin": 0.0,
+        "zero": false,
+        "range":  config.colorScale
+    };
+
+    var scales =  [cScale];
+
+    this.spec.width = config.width;
+    this.spec.height = config.height;
+    this.spec.data = dataTable;
+    this.spec.scales = scales;
+    this.spec.padding = config.padding;
+    this.spec.marks = marks;
+    this.spec.predicates = predicates;
+    this.spec.legends = legends;
+
+};
+
+map.prototype.draw = function(div, callbacks) {
+    var viewUpdateFunction = (function(chart) {
+        this.view = chart({el:div}).renderer(this.config.renderer).update();
+
+        if (callbacks != null) {
+            for (var i = 0; i<callbacks.length; i++) {
+                this.view.on(callbacks[i].type, callbacks[i].callback);
+            }
+        }
+
+    }).bind(this);
+
+    vg.parse.spec(this.spec, viewUpdateFunction);
+};
+
+map.prototype.insert = function(data) {
+
+    var xAxis = this.metadata.names[this.config.x];
+    var yAxis = this.metadata.names[this.config.y];
+    var color = this.metadata.names[this.config.color];
+    var mapType = this.config.mapType;
+    var geoInfoJson = this.config.geoInfoJson;
+
+    for (i = 0; i < data.length; i++) {
+        for (var key in data[i]) {
+            if(key == xAxis){
```

Comment:
Hi Chandana,
We have replace the Vizgrammer.js with Vizgrammer.min.js and refered from common gadget folder, relevant commit is : 
https://github.com/wso2/shared-analytics/pull/126/files#diff-6ac780d61b2b6d470c8ff04bdeed3bcdR52

Thanks,
SajithD